Packs/CiscoAMP/Integrations/CiscoAMP/CiscoAMP.py

```diff
@@ -0,0 +1,3560 @@
+
+    Args:
+        client (Client): Cisco AMP client to run desired requests
+        args (Dict[str, Any]): Arguments passed down by the CLI to provide in the HTTP request.
+
+    Raises:
+        ValueError: Incase the user hasn't entered at least one policy GUID.
+
+    Returns:
+        CommandResults: Information about the updated group.
+    """
+    child_guid = args['child_guid']
+    parent_group_guid = args.get('parent_group_guid')
+
+    raw_response = client.group_parent_update_request(
+        child_guid=child_guid,
+        parent_group_guid=parent_group_guid,
+    )
+
+    context_output = get_context_output(raw_response, ['links'])
+    readable_output = get_readable_output(
+        response=raw_response,
+        header_by_keys=GROUP_HEADERS_BY_KEYS,
+        keys_to_items_option_1=['data'],
+        title=GROUP_TITLE,
+    )
+
+    return CommandResults(
+        outputs_prefix='CiscoAMP.Group',
+        outputs_key_field='guid',
+        outputs=context_output,
+        readable_output=readable_output,
+        raw_response=raw_response,
+    )
+
+
+def group_create_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    """
+    Create a new group and get information about it.
+
+    Args:
+        client (Client): Cisco AMP client to run desired requests
+        args (Dict[str, Any]): Arguments passed down by the CLI to provide in the HTTP request.
+
+    Returns:
+        CommandResults: Information about the new group.
+    """
+    name = args['name']
+    description = args['description']
+
+    raw_response = client.group_create_request(
+        name=name,
+        description=description,
+    )
+
+    context_output = get_context_output(raw_response, ['links'])
+    readable_output = get_readable_output(
+        response=raw_response,
+        header_by_keys=GROUP_HEADERS_BY_KEYS,
+        keys_to_items_option_1=['data'],
+        title=GROUP_TITLE,
+    )
+
+    return CommandResults(
+        outputs_prefix='CiscoAMP.Group',
+        outputs_key_field='guid',
+        outputs=context_output,
+        readable_output=readable_output,
+        raw_response=raw_response,
+    )
+
+
+def groups_delete_command(client: Client, args: Dict[str, Any]) -> CommandResults:
+    """
+    Deletes a group and returns a result if the deletion has succeeded.
+
+    Args:
+        client (Client): Cisco AMP client to run desired requests
+        args (Dict[str, Any]): Arguments passed down by the CLI to provide in the HTTP request.
+
+    Raises:
+        ValueError: Incase the deletion has failed, raise an error.
+
+    Returns:
+        CommandResults: Success message of the deleted group.
+    """
+    group_guid = args['group_guid']
+
+    raw_response = client.group_delete_request(
+        group_guid=group_guid
+    )
+
+    is_deleted = dict_safe_get(raw_response, ['data', 'deleted'])
+
+    if not is_deleted:
+        raise ValueError(f'Failed to delete Group GUID: "{group_guid}".')
```

Comment:
please use `DemistoException`s, and provide `res=raw_response` in all places, to make debugging easier. This attaches the raw response to the exception

Reply:
I've handled all of the `ValueError`s that are after `request` functions, should I also edit the rest of them?

Reply:
pleas pass `res=raw_response` wherever possible

Reply:
Changed every `ValueError` that was after a `request` function.